### PR TITLE
feat(language-server): Support declares and exports in LSP [LNG-304, LNG-319]

### PR DESCRIPTION
**Pull request by DieMyst**
## Description
Allows to get information (definition and type) from export/declares/import tokens

## Checklist
- [x] Corresponding issue has been created and linked in PR title.
- [x] Proposed changes are covered by tests.
- [x] Documentation has been updated to reflect the changes (if applicable).
- [x] I have self-reviewed my code and ensured its quality.
- [x] I have added/updated necessary comments to aid understanding.

## Reviewer Checklist
- [ ] Tests have been reviewed and are sufficient to validate the changes.
- [ ] Documentation has been reviewed and is up to date.
- [ ] Any questions or concerns have been addressed.

### CODE REVIEW

File: semantics/src/main/scala/aqua/semantics/header/ExportSem.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      // Nothing there
+      picker.blank,
+      finSem
+    ).validNec
+  }
+
+  private def finSem(ctx: C, initCtx: C): ValidatedNec[SemanticError[S], C] = {
+
+    val tokens: List[(String, Token[S])] =
+      expr.pubs.map(e => e.fold(n => (n._1.value, n._1), n => (n._1.value, n._1))).toList
```
Comment (InversionSpaces) [inline semantics/src/main/scala/aqua/semantics/header/ExportSem.scala:61]: I don't get it why for `use` and `import` both `name` and `rename` are set as occurrences, but for export only `name` is used.

File: semantics/src/main/scala/aqua/semantics/rules/locations/VariableInfo.scala [hunk 2]
```diff
@@ -3,7 +3,7 @@ package aqua.semantics.rules.locations
 import aqua.parser.lexer.Token
 import aqua.types.Type
 
-case class DefinitionInfo[S[_]](name: String, token: Token[S], `type`: Type)
```
Comment (InversionSpaces) [inline semantics/src/main/scala/aqua/semantics/rules/locations/VariableInfo.scala:6]: Why do we need `isRoot`?

**Comment by DieMyst** [comment]
fixes LNG-304
fixes LNG-319

File: language-server/language-server-api/src/main/scala/aqua/lsp/LocationHandlerLsp.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import aqua.parser.lexer.Token
+import aqua.semantics.header.LocationHandler
+
+class LocationHandlerLsp[S[_]] extends LocationHandler[S, LspContext[S]] {
+
+  override def addOccurences(
+    ctx: LspContext[S],
+    tokens: List[(String, Token[S])],
+    isRoot: Boolean
```
Comment (InversionSpaces) [inline language-server/language-server-api/src/main/scala/aqua/lsp/LocationHandlerLsp.scala:11]: How is it used?

**Pull request merged**